src/v/cluster/tests/leader_balancer_test.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#define BOOST_TEST_MODULE leader_balancer
+
+#include "cluster/scheduling/leader_balancer_greedy.h"
+
+#include <boost/test/unit_test.hpp>
+
+#include <vector>
+
+BOOST_AUTO_TEST_CASE(greedy) {
+    cluster::leader_balancer_strategy::index_type index;
+
+    // 10 nodes, 2 cores -> 20 shards
+    std::vector<model::broker_shard> shards;
+    for (auto n = 0U; n < 10; n++) {
+        for (auto s = 0U; s < 2; s++) {
+            model::broker_shard shard{model::node_id(n), s};
+            shards.push_back(shard);
+        }
+    }
+
+    size_t replica = 0;
+    for (auto shard : shards) {
+        // 10 groups per shard
+        for (auto g = 0U; g < 10; g++) {
+            raft::group_id group(g);
+            // 3 replica per group
+            std::vector<model::broker_shard> replicas;
+            while (replicas.size() != 3) {
+                replicas.push_back(shards[replica++ % shards.size()]);
```

Comment:
ahhh, I guess the greedy algorithm doesn't _enforce_ or _assert_ that. I think its ok since the balancer just looks at load and assumes the data its passed makes sense to whoever created the index. do you see something problematic that I'm not seeing?

Reply:
Nothing explicitly problematic, just suspicious a bit :) I'll leave it up to you, but maybe a small comment that this doesn't matter will help.

Reply:
I changed it so the leader is in each replica set. thanks for pointing this out